modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AbstractAccessControlProvider.java

```diff
@@ -86,16 +86,15 @@ protected void checkForCyclicDependencies(AccessControlGroup group, Set<AccessCo
    */
   protected void collectAccessControls(AccessControlGroup group, Set<AccessControlGroup> toplevelGroups) {
 
-    // TODO hohwille Java HashMap buggy???
```

Comment:
Ok. I just noticed that testProviderCyclic fails if you merely use this code snippet again. This is why I had a look at the hashCode implementation of AccessControlGroup. I changed it as described in the commit message.